Lesson16/ecenaz/script.js

```diff
@@ -0,0 +1,487 @@
+/*
+20. Sum of All Digits in a String
+    - Define a function `sumDigits(str)` that loops through each character of `str`,
+      checks if it's a digit, and if so, adds it to a total sum.
+    - Log the final sum.
+    - Example: "abc123" -> 6
+*/
+function sumDigits(str){
+  let sum=0;
+  for(let i=0; i<str.length; i++){
+    if(str[i]>= '0' && str[i]<='9'){
+      sum += parseInt(str[i]);
+    }
+  }
+  console.log(sum);
+}
+sumDigits("abc123");
+/*
+21. Average of Array Elements
+    - Define a function `averageArray(numbers)` that uses a loop
+      to calculate the average (sum / length).
+    - Log the average.
+*/
+function averageArray(numbers){
+  let sum=0;
+  for (let i=0; i<numbers.length; i++ ){
+sum += numbers[i];
+  }
+  let averageArray = sum / numbers.length;
+  console.log(averageArray);
+}
+averageArray([5,10,15,20,25,30]);
+/*
+22. Flatten a 2D Array (Nested Loops)
+    - Define a function `flattenArray(twoDArray)` that takes an array of arrays
+      (e.g., [[1,2],[3,4]]) and uses nested loops to create a new one-dimensional array.
+    - Log the flattened array.
+*/
+function flattenArray(twoDArray){
+  let flattenedArray= [];
+  for (let i=0; i<twoDArray.length; i++){
+    for (let j=0; j<twoDArray.length; j++){
+      flattenedArray.push(twoDArray[i][j]);
+    }
+  }
+  console.log(flattenedArray);
+}
+
+flattenArray([[1,2],[3,4]]);
+/*
+23. Find Words Containing a Letter
+    - Define a function `findWordsWithLetter(words, letter)` that loops through
+      an array of words and returns a new array of only the words that contain
+      the given letter.
+    - Log the filtered array.
+*/
+function findWordsWithLetter(words, letter){
+  let filteredArray=[];
+for (let i=0; i<words.length; i++){
+  if(words[i].includes(letter)){
+    filteredArray.push(words[i]);
+  }
+}
+ console.log(filteredArray);
+}
+findWordsWithLetter(["elma", "armut","ananas","avokado","muz","turp"],"a");
+
+/*
+24. Push and Pop Operations
+    - Define a function `pushPopExample(arr, itemToPush)` that:
+      - pushes itemToPush to arr
+      - logs the updated array
+      - then pops the last element
+      - logs the popped element
+      - logs the final array
+*/
+function pushPopExample(arr, itemToPush){
+
+  arr.push(itemToPush);
+  console.log("Updated array", arr);
+
+  let poppedItem= arr.pop();
+  console.log("Popped Item:", poppedItem);
+
+  console.log("Final array:", arr);
+}
+
+pushPopExample([1,2,3,4,5],6);
+
+
+/*
+25. Push and Shift Operations
+    - Define a function `manageQueue(queue, newPerson)` that:
+      - push `newPerson` to the end of `queue`
+      - logs the updated queue
+      - shifts (removes) the first person in the queue
+      - logs the removed person
+      - logs the final queue
+*/
+function manageQueue(queue, newPerson){
+  queue.push(newPerson);
+  console.log("Updated queue:", queue);
+
+  let removedPerson = queue.shift();
```

Comment:
The value never changes, therefore better to use const.
```suggestion
  const removedPerson = queue.shift();
```